src/pages/Activity.jsx

```diff
@@ -16,17 +16,25 @@ import {
 export default function Activity() {
   const [isLoading, setIsLoading] = useState(true);
   const [selectedActivity, setSelectedActivity] = useState();
+  const [relatedActivity, setRelatedActivity] = useState();
+
   const { activityUuid } = useParams();
 
   useEffect(() => {
     setIsLoading(true);
     const fetchActivity = async () => {
       try {
         const activity = await fetchActivityByUuid(activityUuid);
+        const singleActivity = await fetchRelatedActivity(activityUuid);
```

Comment:
in questo caso le chiamate dovrebbero essere fatte in parallelo, usando un `Promise.all`
il nome `singleActivity` non rispecchia il risultato effettivo perchè sono una lista di attivit.

C'è da considerare anche il caso in cui non ci siano attività correlate e quindi non si deve visualizzare il carousel